Libraries/Opc.Ua.Server/Subscription/MonitoredItem/MonitoredItemQueue.cs

```diff
@@ -0,0 +1,397 @@
+/* ========================================================================
+ * Copyright (c) 2005-2024 The OPC Foundation, Inc. All rights reserved.
+ *
+ * OPC Foundation MIT License 1.00
+ * 
+ * Permission is hereby granted, free of charge, to any person
+ * obtaining a copy of this software and associated documentation
+ * files (the "Software"), to deal in the Software without
+ * restriction, including without limitation the rights to use,
+ * copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the
+ * Software is furnished to do so, subject to the following
+ * conditions:
+ * 
+ * The above copyright notice and this permission notice shall be
+ * included in all copies or substantial portions of the Software.
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+ * EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES
+ * OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+ * NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT
+ * HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY,
+ * WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+ * FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+ * OTHER DEALINGS IN THE SOFTWARE.
+ *
+ * The complete license agreement can be found here:
+ * http://opcfoundation.org/License/MIT/1.00/
+ * ======================================================================*/
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Newtonsoft.Json.Linq;
+
+namespace Opc.Ua.Server
+{
+    /// <summary>
+    /// A factory for <see cref="IDataChangeMonitoredItemQueue"> and </see> <see cref="IEventMonitoredItemQueue"/>
+    /// </summary>
+    public class MonitoredItemQueueFactory : IMonitoredItemQueueFactory
+    {
+        /// <inheritdoc/>
+        public bool SupportsDurableQueues => false;
+        /// <inheritdoc/>
+        public IDataChangeMonitoredItemQueue CreateDataChangeQueue(bool createDurable)
+        {
+            return new DataChangeMonitoredItemQueue(createDurable);
+        }
+
+        /// <inheritdoc/>
+        public IEventMonitoredItemQueue CreateEventQueue(bool createDurable)
+        {
+            return new EventMonitoredItemQueue(createDurable);
+        }
+
+        /// <inheritdoc/>
+        public void Dispose()
+        {
+            //only needed for managed resources
+        }
+    }
+    /// <summary>
+    /// Provides a queue for data changes.
+    /// </summary>
+    public class DataChangeMonitoredItemQueue : IDataChangeMonitoredItemQueue
```

Comment:
for new code you could also create new .cs files per class

Reply:
done, where should I put the interfaces? separate files, one file for all 3, below the implementations?